platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/FilteredFileCollection.java

```diff
@@ -78,4 +79,25 @@ public boolean contains(File file) {
     public Iterator<File> iterator() {
         return Iterators.filter(collection.iterator(), filterSpec::isSatisfiedBy);
     }
+
+    @Override
+    public Optional<FileCollectionExecutionTimeValue> calculateExecutionTimeValue() {
+        Optional<FileCollectionExecutionTimeValue> sourceExecutionTimeValue = collection.calculateExecutionTimeValue();
+        return sourceExecutionTimeValue.map(fileCollectionExecutionTimeValue -> new FilteredExecutionTimeValue(fileCollectionExecutionTimeValue, filterSpec));
+    }
+
+    private static class FilteredExecutionTimeValue implements FileCollectionExecutionTimeValue {
+        private final FileCollectionExecutionTimeValue sourceExecutionTimeValue;
```

Comment:
💅 The scope of the name is small and specific enough that a shorter name like `source` is an appropriate name